dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -0,0 +1,417 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/webdataset_loader.h"
+#include <cstddef>
+#include <cstring>
+#include <limits>
+#include <memory>
+#include <utility>
+#include "dali/core/error_handling.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/util/file.h"
+
+namespace dali {
+namespace detail {
+namespace wds {
+
+inline MissingExtBehavior Str2MissingExt(std::string missing_component_behavior) {
+  std::remove(missing_component_behavior.begin(), missing_component_behavior.end(), '_');
+  std::transform(missing_component_behavior.begin(), missing_component_behavior.end(),
+                 missing_component_behavior.begin(), static_cast<int (*)(int)>(std::tolower));
+  if (missing_component_behavior == "") {
+    return MissingExtBehavior::Empty;
+  } else if (missing_component_behavior == "skip") {
+    return MissingExtBehavior::Skip;
+  } else if (missing_component_behavior == "fillempty") {
+    return MissingExtBehavior::Empty;
+  } else if (missing_component_behavior == "raise") {
+    return MissingExtBehavior::Raise;
+  } else if (missing_component_behavior == "raiseerror") {
+    return MissingExtBehavior::Raise;
+  } else {
+    return MissingExtBehavior::Invalid;
+  }
+}
+
+inline SampleConfig ParseSampleConfig(std::ifstream& config, const std::string& config_path) {
+  SampleConfig out;
+  config >> out.start_offset;
+  out.end_offset = std::numeric_limits<decltype(out.end_offset)>::max();
+
+  std::string component_metadata;
+  std::getline(config, component_metadata);
+  component_metadata += ' ';
```

Comment:
This doesn't seem necessary. See:
https://godbolt.org/z/j3s8v3zPY

Reply:
alright